config/config-api/src/main/java/com/thoughtworks/go/config/remote/ConfigRepoConfig.java

```diff
@@ -142,29 +88,8 @@ private void validateUniquenessOfId(ValidationContext validationContext) {
         }
     }
 
-    public boolean validateTree(ValidationContext validationContext) {
-        validate(validationContext);
-        boolean isValid = errors.isEmpty();
-
-        if (getMaterialConfig() != null) {
-            isValid = getMaterialConfig().errors().isEmpty() && isValid;
```

Comment:
@kritika-singh3 where has this check moved? This check ensured a config repo is invalid if the material config within is invalid.

Reply:
@maheshp The check for `materialConfig` is present in `validate(ValidationContext )` [method](https://github.com/gocd/gocd/blob/master/config/config-api/src/main/java/com/thoughtworks/go/config/remote/ConfigRepoConfig.java#L55-L65). The same will get called from the `validateTree` [method](https://github.com/gocd/gocd/blob/master/config/config-api/src/main/java/com/thoughtworks/go/config/rules/RuleAwarePluginProfile.java#L127-L136) from `RuleAwarePluginProfile` class.